lib/logitech_receiver/base.py

```diff
@@ -189,6 +189,17 @@ def read(handle, timeout=DEFAULT_TIMEOUT):
 		return reply[1:]
 
 
+# sanity checks on  message and report incorrect sizes when debugging
+def debug_message(data) :
+	assert isinstance(data, bytes), (repr(data), type(data))
+	report_id = ord(data[:1])
+	assert ((report_id & 0xF0 == 0) or
+			(report_id == 0x10 and len(data) == _SHORT_MESSAGE_SIZE) or
+			(report_id == 0x11 and len(data) == _LONG_MESSAGE_SIZE) or
+			(report_id == 0x20 and len(data) == _MEDIUM_MESSAGE_SIZE)), \
+			"unexpected message size: report_id %02X message %s" % (report_id, _strhex(data))
```

Comment:
I think we should create a dictionary for the report lenght.

```python
report_lenght = {
	0x10: 7
	0x11: 20
	...
}
```

then

```python
assert report_id in report_lenght and len(data) == report_lenght[report_id]
```

Reply:
OK.   Also (report_id & 0xF0 == 0) isn't a possibility as far as I can tell.  All reports should be 0x10, 0x11, or 0x20.  Or am I mistaken?  (Maybe non-DJ device reports fall into this category?)

Reply:
I don't really like that approach as it is not really clear what `0xF0` is when we read the code.

Reply:
With the change  suggested by @Lekensteyn other report ids will fail the check (and they would have failed the check before) so this branch of the check can be removed.

Reply:
I agree.  There are old-style (non-DJ) reports with report ids not 10,11,20,21.  This is what the code is checking for.  I put in a constant for the max report id for these reports and modified the checks to check against that instead.